### PR TITLE
Thrift udf api

### PR DESCRIPTION
**Pull request by rongrong**
Test plan - travis

```
== RELEASE NOTES ==

Thrift Connector Changes
* Rename ``presto-thrift-connector-api`` to ``presto-thrift-api`` and have separate packages for datatypes, valuesets and connector.
```

### CODE REVIEW

File: presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/PrestoThriftUdfService.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.facebook.drift.TException;
+import com.facebook.drift.annotations.ThriftField;
+import com.facebook.drift.annotations.ThriftMethod;
+import com.facebook.drift.annotations.ThriftService;
+import com.google.common.util.concurrent.ListenableFuture;
+
+@ThriftService(value = "presto-udf", idlName = "PrestoThriftUdfService")
+public interface PrestoThriftUdfService
+{
+    @ThriftMethod("prestoInvokeUdf")
```
Comment (caithagoras) [inline presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/PrestoThriftUdfService.java:25]: Why not just `@ThriftMethod` without `("prestoInvokeUdf")`

File: presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/PrestoThriftUdfService.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    ListenableFuture<PrestoThriftUdfResult> invokeUdf(
+            @ThriftField(name = "functionHandle") PrestoThriftFunctionHandle functionHandle,
+            @ThriftField(name = "inputs") PrestoThriftUdfPage inputs)
```
Comment (caithagoras) [inline presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/PrestoThriftUdfService.java:28]: `(name = )` is unnecessary IIUC.

File: presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/PrestoThriftUdfService.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            throws PrestoThriftUdfServiceException, TException;
```
Comment (caithagoras) [inline presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/PrestoThriftUdfService.java:29]: Usually we do
```
throws TException,
@ThriftId(1) PrestoThriftUdfServiceException;
```

File: presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/ThriftUdfPage.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+@ThriftStruct("UdfPage")
+public class ThriftUdfPage
+{
+    private final ThriftUdfPageFormat pageFormat;
+    private final List<PrestoThriftBlock> thriftBlocks;
+
+    @ThriftConstructor
+    public ThriftUdfPage(
+            ThriftUdfPageFormat pageFormat,
+            @Nullable List<PrestoThriftBlock> thriftBlocks)
```
Comment (caithagoras) [inline presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/ThriftUdfPage.java:38]: Let's use `Optional` instead of `@Nullable`.

Comment (rongrong) [reply presto-thrift-api/src/main/java/com/facebook/presto/thrift/api/udf/ThriftUdfPage.java:38]: Are they going to be translated to the same thrift idl?

**Pull request merged**